### DISCUSSION TITLE
The invalidation of pysr under some constraint conditions

### DISCUSSION THREAD

**Discussion by jackeuylov** [MilesCranmer/PySR#743] [Q&A]
Hi Miles @MilesCranmer
In my work, I need to lock the equation y1 = f( k1, k2,... ) * x1 + g( k1, k2,... ) * x2, (x, y, k are all variables) , and need to balance the dimension, and limit the appearance of constants. Under these constraints, pysr unexpectedly fails : it cannot grow a reliable expression tree, but stays at the root node. 
I have tried to increase randomness and increase the number of populations, but all failed. As shown in the picture, the final result has only a very simple expression.
<img width="592" alt="27ae1f3640dc0490cf4c4c5fbac98cd" src="https://github.com/user-attachments/assets/9f40b0b2-7365-407d-a7fa-6f791992e7c9">
But miraculously, when I reverse the problem, that is, x1 = f ( k1, k2,... ) * y1 + g ( k1, k2,... ) * y2, pysr can grow normally and evolve a binary tree.
<img width="669" alt="ab3d23969d64f2b77de1f4332a5290a" src="https://github.com/user-attachments/assets/4e2e4a77-40b5-4245-9168-ab6b50429d9d">
Do you have any clue about this ? I guess it is because the amount of information contained in x and y is different ?
Thanks！

**Comment by MilesCranmer** [comment]
This is going to be so so so easy once 1.0 drops. This kind of feature is built-in! 1.0.0 should drop this or next week.

but until then, what is your current approach? Is it a custom loss function?

**Reply by jackeuylov** [reply]
Yes, I implemented the built-in function according to[ Example 9](https://astroautomata.com/PySR/examples/#9-custom-objectives), which returns the maximum value for a binary tree that does not satisfy the specified equation form. I have been trying to denoise the data for the past few days, but it still doesn't seem to work. I suspect that it may be because there is no solution in the solution space of this form? This is the only possibility I can think of, so I am planning to abandon this equation form.
However, the built-in functions I customized for controlling other equation forms have been working very well.

**Reply by MilesCranmer** [reply]
Check out https://astroautomata.com/SymbolicRegression.jl/dev/examples/template_expression/
